packages/@n8n/task-runner/src/data-request/data-request-response-reconstruct.ts

```diff
@@ -10,18 +10,41 @@ export class DataRequestResponseReconstruct {
 	/**
 	 * Reconstructs `connectionInputData` from a DataRequestResponse
 	 */
-	reconstructConnectionInputData(
+	reconstructConnectionInputItems(
 		inputData: DataRequestResponse['inputData'],
+		chunk?: InputDataChunkDefinition,
 	): INodeExecutionData[] {
-		return inputData?.main?.[0] ?? [];
+		const inputItems = inputData?.main?.[0] ?? [];
+		if (!chunk) {
+			return inputItems;
+		}
+
+		// Only a chunk of the input items was requested. We reconstruct
+		// the array by filling in the missing items with `undefined`.
+		let sparseInputItems: Array<INodeExecutionData | undefined> = [];
+
+		sparseInputItems = sparseInputItems
+			.concat(Array.from({ length: chunk.startIndex }))
+			.concat(inputItems)
+			.concat(Array.from({ length: inputItems.length - chunk.startIndex - chunk.count }));
+
+		return sparseInputItems as INodeExecutionData[];
```

Comment:
The assertion is misrepresenting what this is. Would it be possible to use `Array<INodeExecutionData | undefined>`? As that'd be more accurate

Reply:
It's intentional. I moved it to another location and added comment